document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -985,7 +985,11 @@ protected Document prepareDocument() throws SQLException, IOException {
       Map<String, Object> jsonNode = new HashMap();
       for (int i = 1; i <= columnCount; i++) {
         String columnName = resultSetMetaData.getColumnName(i);
-        String columnValue = resultSet.getString(i);
+        int columnType = resultSetMetaData.getColumnType(i);
+        String columnValue =
+            columnType == 2003
```

Comment:
Why is this required?
And, can we replace this magic number (2003) with a descriptive variable name?

Reply:
I tried looking for it - `resultSetMetaData.getColumnType(i)` - for corresponding constant expose as string. But, didn't find any enums, so have to hard-code. will look again.

Reply:
Added comment here and todo.